easybuild/easyblocks/generic/cmakemake.py

```diff
@@ -182,32 +196,49 @@ def configure_step(self, srcdir=None, builddir=None):
                 print_warning('Ignoring BUILD_SHARED_LIBS is set in configopts because build_shared_libs is set')
             self.cfg.update('configopts', '-DBUILD_SHARED_LIBS=%s' % ('ON' if build_shared_libs else 'OFF'))
 
-        env_to_options = {
-            'CC': 'CMAKE_C_COMPILER',
-            'CFLAGS': 'CMAKE_C_FLAGS',
-            'CXX': 'CMAKE_CXX_COMPILER',
-            'CXXFLAGS': 'CMAKE_CXX_FLAGS',
-            'F90': 'CMAKE_Fortran_COMPILER',
-            'FFLAGS': 'CMAKE_Fortran_FLAGS',
-        }
+        # If the cache does not exist CMake reads the environment variables
+        cache_exists = os.path.exists('CMakeCache.txt')
```

Comment:
Does the `CMakeCache.txt` know what system it is for? So, if we have `separate_build_dir = False` and the developers have added this file to the code could we end up using a CMake Cache for a different system?

Reply:
This file is basically the output of "configure", so developers must not add this file.
However below I added code to handle the case where this file exists by setting CMake options with `-D` which overwrites cache entries. So from our side here there is no functional difference even for the case where this file exists (again: which it shouldn't)